exchange/bitswap/decision/peer_request_queue.go

```diff
@@ -198,12 +266,20 @@ type activePartner struct {
 
 	// priority queue of tasks belonging to this peer
 	taskQueue pq.PQ
+
+	//  kinda dirty to give each active partner all the ledgers and it's pid
+	//  instead of just its own ledger, but i'm not sure how else to deal with
+	//  the lazily loaded ledgers
```

Comment:
the avantage is not "giv[ing] each active partner all the ledgers" but instead a pointer to a common set, so state updates in one place.